src/__integrations__/pmClient.checkStatus.integration.test.ts

```diff
@@ -25,13 +28,32 @@ describe('Payment Manager Client', () => {
   // eslint-disable-next-line functional/no-let
   let paymentManagerClient;
 
+  // eslint-disable-next-line functional/no-let
+  let paymentManagerClientWithPolling;
+
+  const transientErrorGivenFinalStatusCondition = async (r: Response): Promise<boolean> => {
+    const myJson = (await r.clone().json()) as TransactionStatusResponse;
```

Comment:
@infantesimone _cloning_ the response is still needed after fixing polling logic?

Reply:
You have to clone the response to use the body in the custom condition, because the body is used twice:
- the first time in the custom condition that is passed to the polling fetch;
- the second time in the implementation of the underlying fetch (to build the final result).